micronota/bfillings/prodigal.py

```diff
@@ -0,0 +1,136 @@
+# ----------------------------------------------------------------------------
+# Copyright (c) 2015--, micronota development team.
+#
+# Distributed under the terms of the Modified BSD License.
+#
+# The full license is in the file COPYING.txt, distributed with this software.
+# ----------------------------------------------------------------------------
+
+from os import makedirs
+from os.path import join
+from burrito.parameters import FlagParameter, ValuedParameter
+from burrito.util import CommandLineApplication, ResultPath
+
+
+class Prodigal(CommandLineApplication):
+    '''Prodigal (version 2.6.2) application controller.
+    '''
+    _command = 'prodigal'
+    _valued_path_options = [
+        # Specify FASTA/Genbank input file (default reads from stdin).
+        '-i',
+        # Write protein translations to the selected file.
+        '-a',
+        # Write nucleotide sequences of genes to the selected file.
+        '-d',
+        # Write all potential genes (with scores) to the selected file.
+        '-s',
+        # Write a training file (if none exists);
+        # otherwise, read and use the specified training file.
+        '-t',
+        # Specify output file (default writes to stdout).
+        '-o',
+    ]
+    _valued_nonpath_options = [
+        # Select output format (gbk, gff, or sco).  Default is gbk.
+        '-f',
+        # Specify a translation table to use (default 11).
+        '-g',
+        # Treat runs of N as masked sequence; don't build genes across them.
+        '-m',
+        # Select procedure (single or meta).  Default is single.
+        '-p',
+    ]
+    _flag_options = [
+        # Closed ends.  Do not allow genes to run off edges.
+        '-c',
+        # Print version number and exit.
+        '-v',
+        # Run quietly (suppress normal stderr output).
+        '-q',
+        # Print help menu and exit.
+        '-h',
+        # Bypass Shine-Dalgarno trainer and force a full motif scan.
+        '-n',
+    ]
+
+    _parameters = {}
+    _parameters.update({
+        i: ValuedParameter(
+            Prefix=i[0], Name=i[1:], Delimiter=' ',
+            IsPath=True)
+        for i in _valued_path_options})
+    _parameters.update({
+        i: ValuedParameter(
+            Prefix=i[0], Name=i[1:], Delimiter=' ')
+        for i in _valued_nonpath_options})
+    _parameters.update({
+        i: FlagParameter(
+            Prefix=i[0], Name=i[1:])
+        for i in _flag_options})
+    _suppress_stderr = False
+
+    def _accept_exit_status(self, exit_status):
+        if exit_status == 0:
+            return True
+        else:
+            return False
+
+    def _get_result_paths(self, data):
```

Comment:
this could be expressed as:

```python
def _get_result_paths(self, data):
    ison_and_values = ((self.Parameters[i].isOn(), self.Parameters[i].Value)
                        for i in self._valued_path_options if i != '-i')
    return {o: ResultPath=self._absolute(v) for o, v in ison_and_values}
```

...though not sure if that is an improvement